### PR TITLE
[APCD] Extension Form Edit Modal -  Expected Date error 

### PR DESCRIPTION
**Pull request by chandra-tacc**
## Overview
1. Use the applicable expected list of applicable values. With drop down, remove the chance of bad values from user.
2. Map the value to the available value - use date conversion from label to api value

## Related

[WP-768](https://tacc-main.atlassian.net/browse/WP-768)

## Testing

1. Use edit modal and adjust dates: http://localhost:8000/administration/list-extensions/

## UI
Screenshot 2024-11-15 at 11 37 29 AM

### CODE REVIEW

File: apcd-cms/src/client/src/components/Extensions/EditExtensionModal/EditExtensionModal.tsx [hunk 1]
```diff
@@ -193,190 +201,207 @@ const EditExtensionModal: React.FC<EditExtensionModalProps> = ({
...
-                      id="ext_status"
-                      onChange={formik.handleChange}
-                      onBlur={formik.handleBlur}
-                      className={`form-control ${styles.viewRecord}`}
-                      value={formik.values.ext_status}
-                    >
-                      {statusOptions?.map(
-                        (opt) =>
-                          opt.value !== 'All' && (
+          <QueryWrapper
```
Comment (chandra-tacc) [inline apcd-cms/src/client/src/components/Extensions/EditExtensionModal/EditExtensionModal.tsx:204]: Lot of formatting but only change is to use QueryWrapper.

File: apcd-cms/src/apps/utils/apcd_database.py [hunk 2]
```diff
@@ -1023,10 +1023,10 @@ def update_extension(form):
         set_values = []
         # to set column names for query to the correct DB name
         columns = {
-            'applicable-data-period': 'applicable_data_period',
+            'applicable_data_period': 'applicable_data_period',
```
Comment (chandra-tacc) [inline apcd-cms/src/apps/utils/apcd_database.py:1026]: removed the confusion with dash vs underscore and made everything the same.

File: apcd-cms/src/client/src/utils/dateUtil.ts [hunk 3]
```diff
@@ -13,3 +13,45 @@ export const formatDate = (dateString: string | number | Date): string => {
     hour12: true,
   }).format(date);
 };
+
+/**
+ *
+ * @param period string - Jan. 2024 as example
+ * @returns string. 2024-01 for Jan. 2024.
+ */
+export const convertPeriodLabelToApiValue = (period: string): string | null => {
```
Comment (chandra-tacc) [inline apcd-cms/src/client/src/utils/dateUtil.ts:22]: needed this because the api was returning Jan. 2024 which does not fit with db values

**Review by happycodemonkey — APPROVED** [review]
This looks great!

**Pull request merged**